src/components/Sidebar.tsx

```diff
@@ -59,29 +66,58 @@ export interface ISidebarNodeProps extends INode {
   divider?: boolean;
 }
 
-type SidebarItemProps = {
+interface IPanningActions {
+  enableDrag?: () => void;
+  disableDrag?: () => void;
+}
+interface SidebarItemProps extends IPanningActions {
   node: INode | ISidebarNodeProps;
   selected?: boolean;
   updateSelected?: (id: string) => void;
-};
+}
 
+export interface ISubSidebarItemProps {
+  node: INode | ISidebarNodeProps;
+  selected?: boolean;
+  updateSelected?: (id: string) => void;
+}
 export interface ISidebarProps {
   selectedBarNode?: string;
   sidebarNodes?: ISidebarNodeProps[];
   updateSelectedBar?: (id: string) => void;
+  engine: CanvasEngine;
+}
+
+interface IHandleClick extends IPanningActions {
+  event: unknown;
+  node: INode | ISidebarNodeProps;
+  updateSelected?: (id: string) => void;
+}
+
+interface ISubSidebar {
+  show?: boolean;
+  nodes?: INode[];
 }
 
 // handle item click callback
-const handleItemClick = (
-  event: unknown,
-  node: INode | ISidebarNodeProps,
-  updateSelected?: (id: string) => void
-) => {
+const handleItemClick = ({
+  event,
+  node,
+  updateSelected,
+  enableDrag,
+  disableDrag,
+}: IHandleClick) => {
   const { id, draggable, preCallback, postCallback, defaultCallback } = node;
   // if item is un-draggable click event fires only
   if (!draggable) {
     if (!!updateSelected) updateSelected(id);
 
+    // execute drag actions
+    if (id.startsWith(DefaultSidebarNodeTypes.PANNING) && enableDrag)
```

Comment:
As a minor detail, I would prefer to use an equality check instead of the startsWith. The rationality being that I don't believe is useful to allow multiple sidebar buttons to enable/disable the drag.